### ISSUE TITLE
About Creating Solids 

### ISSUE THREAD

**Issue by nickyang99** [tanmayms3009/autocad_automation#1]
I don't know how to use this method. Can you help me？
RetVal = object.AddExtrudedSolidAlongPath(Profile, Path)

**Comment by tanmayms3009** [comment]
> I don't know how to use this method. Can you help me？ RetVal = object.AddExtrudedSolidAlongPath(Profile, Path)

Hi, to use this method you need to store the region in a variable and then pass the same as a parameter for "profile".  You have to pass the Polyline, Circle, Ellipse, Spline, or Arc as a "path" parameter to create an object along that path. Also, you need to create a block of the object that you want to extrude and use this method against the same object. You can see my blogs to find other solutions of pyautocad: https://www.supplychaindataanalytics.com/author/tanmaysawant/

**Comment by nickyang99** [comment]
I have watched your blog about Solid objects, but it lacks the method of stretching along the path. And I have tried to pass in two parameters as required, but he still failed. Is there something wrong with the explanation given by the official?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tanmayms3009/autocad_automation"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年4月20日(星期三) 晚上10:25
***@***.***&gt;;
***@***.******@***.******@***.***&gt;;
主题:&nbsp;Re: [tanmayms3009/autocad_automation] About Creating Solids  (Issue #1)





  
I don't know how to use this method. Can you help me？ RetVal = object.AddExtrudedSolidAlongPath(Profile, Path)
  
Hi, to use this method you need to store the region in a variable and then pass the same as a parameter for "profile".  You have to pass the Polyline, Circle, Ellipse, Spline, or Arc as a "path" parameter to create an object along that path. Also, you need to create a block of the object that you want to extrude and use this method against the same object. You can see my blogs to find other solutions of pyautocad: https://www.supplychaindataanalytics.com/author/tanmaysawant/

**Comment by tanmayms3009** [comment]
Thanks for reaching out to us through our blog.

The official document is not sufficient for coding certain things. According to my knowledge, we must try the following steps.

Firstly any Add method for drawing geometries like AddCircle, AddLine, AddPolyline, etc returns us <class 'win32com.client.CDispatch'> . This “CDispatch” has to be converted to “IDispatch” so that we can create an Array of Dispatch type of elements to form a region using VT_DISPATCH objects.
Secondly, the profile (region) and the path have to be in two different planes from each other.
After achieving the above-mentioned points we can draw an Extruded Solid by using the AddExtrudedSolidAlongPath method.

Hope this opens up some doors for you.

**Comment by nickyang99** [comment]
Thanks for your reply,what's the meaning of&nbsp; ''This “CDispatch” has to be converted to “IDispatch”''?
Can you show me how to do it




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tanmayms3009/autocad_automation"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年4月24日(星期天) 中午1:48
***@***.***&gt;;
***@***.******@***.******@***.***&gt;;
主题:&nbsp;Re: [tanmayms3009/autocad_automation] About Creating Solids  (Issue #1)





 
Thanks for reaching out to our blog.
 
The official document is not sufficient for coding certain things. According to my knowledge, we must try the following steps.
 
Firstly any Add method for drawing geometries like AddCircle, AddLine, AddPolyline, etc returns us <class 'win32com.client.CDispatch'&gt; . This “CDispatch” has to be converted to “IDispatch” so that we can create an Array of Dispatch type of elements to form a region using VT_DISPATCH objects.
 Secondly, the profile (region) and the path have to be in two different planes from each other.
 After achieving the above-mentioned points we can draw an Extruded Solid by using the AddExtrudedSolidAlongPath method.
 
Hope this opens up some doors for you.

**Comment by tanmayms3009** [comment]
Basically we can create arrays of different data types using win32com library if the same task is not possible to complete just with the pyautocad library. Check this page out for more information: https://docs.microsoft.com/en-us/windows/win32/api/wtypes/ne-wtypes-varenum

Actually I am trying to accomplish the same task as per your request. I will spend some time on that whenever I get time and will definitely post the solution as soon as I find the same.

**Comment by nickyang99** [comment]
I still failed to achieve RetVal = object. AddExtrudedSolidAlongPath (Profile, Path), I don't know how your progress, this is my code and an error, can you give me some advice?








------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tanmayms3009/autocad_automation"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年4月24日(星期天) 晚上10:57
***@***.***&gt;;
***@***.******@***.******@***.***&gt;;
主题:&nbsp;Re: [tanmayms3009/autocad_automation] About Creating Solids  (Issue #1)





 
Basically we can create arrays of different data types using win32com library if the same task is not possible to complete just with the pyautocad library. Check this page out for more information: https://docs.microsoft.com/en-us/windows/win32/api/wtypes/ne-wtypes-varenum
 
Actually I am trying to accomplish the same task as per your request. I will spend some time on that whenever I get time and will definitely post the solution as soon as I find the same.

**Comment by Marekej** [comment]
Hi, is there any possibility to extract area from object (for inst. polygon)? Thank you.
